src/core/IronPython.Modules/signal.cs

```diff
@@ -417,35 +418,97 @@ private static void SetPythonSignalState(CodeContext/*!*/ context, PythonSignalS
                 foreach (int sig in sigs) {
                     PySignalToPyHandler[sig] = sig_dfl;
                 }
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux)) {
+                    for (int sig = SIGRTMIN; sig <= SIGRTMAX; sig++) {
+                        PySignalToPyHandler[sig] = sig_dfl;
+                    }
+                }
+
+                // Setting exceptions to the rule
                 PySignalToPyHandler[SIGINT] = default_int_handler;
                 if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux) || RuntimeInformation.IsOSPlatform(OSPlatform.OSX)) {
                     PySignalToPyHandler[SIGPIPE] = sig_ign;
                     PySignalToPyHandler[SIGXFSZ] = sig_ign;
                 }
                 if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX)) {
-                    PySignalToPyHandler.Remove(SIGKILL);
-                    PySignalToPyHandler.Remove(SIGSTOP);
+                    PySignalToPyHandler[SIGKILL] = null;
+                    PySignalToPyHandler[SIGSTOP] = null;
+                }
+            }
+
+
+            public virtual bool TryGetPyHandler(int signalnum, out object? value)
+                => (value = PySignalToPyHandler[signalnum]) != null;
+
+
+            public virtual void SetPyHandler(int signalnum, object value)
+                => PySignalToPyHandler[signalnum] = value;
+
+
+            /// <summary>
+            /// Call the Python handler callable passing the given signal number as argument to the callable.
+            /// </summary>
+            protected void CallPythonHandler(int signum, object? handler) {
+                if (handler is null) return;
+
+                if (handler == default_int_handler) {
+                    // We're dealing with the default_int_handlerImpl which we
+                    // know doesn't care about the frame parameter
+                    default_int_handlerImpl(signum, null);
+                    return;
+                } else {
+                    // We're dealing with a callable matching PySignalHandler's signature
+                    try {
+                        PySignalHandler temp = (PySignalHandler)Converter.ConvertToDelegate(handler,
+                                                                                            typeof(PySignalHandler));
+
+                        if (SignalPythonContext.PythonOptions.Frames) {
+                            temp.Invoke(signum, SysModule._getframeImpl(null,
+                                                                        0,
+                                                                        SignalPythonContext._mainThreadFunctionStack));
+                        } else {
+                            temp.Invoke(signum, null);
+                        }
+                    } catch (Exception ex) {
+                        System.Console.WriteLine(SignalPythonContext.FormatException(ex));
```

Comment:
Console.WriteLine is a bit weird, but I guess that was there before... so LGTM!

Reply:
Yes, it is weird and yes, it was like that before. It seems that CPython prints it on stderr and then terminates.